crates/blockchain/blockchain.rs

```diff
@@ -98,6 +90,7 @@ pub fn validate_state_root(
     }
 }
 
+// Returns the hash of the head of the canonical chain (the latest valid hash).
```

Comment:
nit: maybe we should rename the function to `latest_canonical_block_hash` then?

Reply:
Done.